### ISSUE TITLE
PCA outputting infinite value result in Cluster failed!

### ISSUE THREAD

**Issue by baosongliang** [ZPDu/Domain-general-Crowd-Counting-in-Unseen-Scenarios#3]
Thanks for your great work! When I try to train from scratch use SHA dataset, I got PCA outputting infinite values. Use this PCA results to do Kmeans got error as follows:

ValueError: Input X contains infinity or a value too large for dtype('float32').
![err](https://user-images.githubusercontent.com/12793911/218663602-e7187041-9365-44a8-a2c5-a3f347273a55.png)

**Comment by ZPDu** [comment]
> Thanks for your great work! When I try to train from scratch use SHA dataset, I got PCA outputting infinite values. Use this PCA results to do Kmeans got error as follows:
> 
> ValueError: Input X contains infinity or a value too large for dtype('float32'). ![err](https://user-images.githubusercontent.com/12793911/218663602-e7187041-9365-44a8-a2c5-a3f347273a55.png)

Thanks for your interest to our work and for reaching out! I just checked the uploaded code, also tried to train from scratch, and it seems nothing wrong to me. Could you please use the processed dataset in https://github.com/gjy3035/C-3-Framework? I will also add this when updating the readme file.

**Comment by baosongliang** [comment]
> > Thanks for your great work! When I try to train from scratch use SHA dataset, I got PCA outputting infinite values. Use this PCA results to do Kmeans got error as follows:
> > ValueError: Input X contains infinity or a value too large for dtype('float32'). ![err](https://user-images.githubusercontent.com/12793911/218663602-e7187041-9365-44a8-a2c5-a3f347273a55.png)
> 
> Thanks for your interest to our work and for reaching out! I just checked the uploaded code, also tried to train from scratch, and it seems nothing wrong to me. Could you please use the processed dataset in https://github.com/gjy3035/C-3-Framework? I will also add this when updating the readme file.

Thanks for your reply! I've tried use the processed data mentioned in https://github.com/gjy3035/C-3-Framework and download the processed data from [this link](https://mailnwpueducn-my.sharepoint.com/personal/gjy3035_mail_nwpu_edu_cn/_layouts/15/onedrive.aspx?ga=1&id=%2Fpersonal%2Fgjy3035%5Fmail%5Fnwpu%5Fedu%5Fcn%2FDocuments%2F%E8%AE%BA%E6%96%87%E5%BC%80%E6%BA%90%E6%95%B0%E6%8D%AE%2FC3Data). But still got the same issue, could you pls use the linked url data and try again?

**Comment by ZPDu** [comment]
> > > Thanks for your great work! When I try to train from scratch use SHA dataset, I got PCA outputting infinite values. Use this PCA results to do Kmeans got error as follows:
> > > ValueError: Input X contains infinity or a value too large for dtype('float32'). ![err](https://user-images.githubusercontent.com/12793911/218663602-e7187041-9365-44a8-a2c5-a3f347273a55.png)
> > 
> > 
> > Thanks for your interest to our work and for reaching out! I just checked the uploaded code, also tried to train from scratch, and it seems nothing wrong to me. Could you please use the processed dataset in https://github.com/gjy3035/C-3-Framework? I will also add this when updating the readme file.
> 
> Thanks for your reply! I've tried use the processed data mentioned in https://github.com/gjy3035/C-3-Framework and download the processed data from [this link](https://mailnwpueducn-my.sharepoint.com/personal/gjy3035_mail_nwpu_edu_cn/_layouts/15/onedrive.aspx?ga=1&id=%2Fpersonal%2Fgjy3035%5Fmail%5Fnwpu%5Fedu%5Fcn%2FDocuments%2F%E8%AE%BA%E6%96%87%E5%BC%80%E6%BA%90%E6%95%B0%E6%8D%AE%2FC3Data). But still got the same issue, could you pls use the linked url data and try again?

Hi, sorry but it seems that I don't have access to your link. Could you please try to directly use the processed dataset in C-3 framework? Also I've updated new information for the preparation step, let me know if there are any problems.

**Comment by baosongliang** [comment]
@ZPDu 
I found the reason why got inf value. As pic below, if I use npdata.astype('float32') and do pca.fit_transform(npdata), the result will got overflowed data, and result in +inf and -inf value. I changeed npdata.astype('float32') to npdata.astype('float64') then worked fine!
![1677209489023](https://user-images.githubusercontent.com/12793911/221085330-29a3799a-1145-436c-a292-90d92a209b0d.png)
